### PR TITLE
LPS-96358 - As a Search Admin, I want to enable and disable rankings

### PR DESCRIPTION
**Pull request by thektan**
https://issues.liferay.com/browse/LPS-96358

Hopefully these changes should be straightfoward. I'm mainly passing and reading in an `inactive` parameter based on existing logic. Values on the edit page are still being read in from URL params (https://github.com/BryanEngler/liferay-portal/pull/435/files#diff-11f81148884c3be770846605eb905542R81), but maybe in a separate ticket we can switch to retrieving the ranking based on ID.

I'm also not sure if I did this part properly where I'm including inactive rankings while checking for duplicates 951fde39b8599b78b16823d1bb37a2192a4c7d39. So when duplicates are allowed, it'll still check to make sure only 1 ranking is active.

Ran tests at https://github.com/thektan/liferay-portal/pull/78#issuecomment-551023615, but looks like a lot of things timed out, so might be worth running again.

Thanks again!

### CODE REVIEW

**Comment by thektan** [comment]
ci:test:search

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/main/java/com/liferay/portal/search/tuning/rankings/web/internal/index/DuplicateQueryStringsDetectorImpl.java [hunk 1]
```diff
@@ -77,10 +79,18 @@ protected BooleanQuery getCriteriaQuery(Criteria criteria) {
 		_addQueryClauses(
 			booleanQuery::addFilterQueryClauses, getQueryStringsQuery(criteria),
 			getIndexQuery(criteria));
-		_addQueryClauses(
-			booleanQuery::addMustNotQueryClauses,
-			queries.term(RankingFields.INACTIVE, true),
-			getUnlessRankingIdQuery(criteria));
+
```
Comment (BryanEngler) [inline modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/main/java/com/liferay/portal/search/tuning/rankings/web/internal/index/DuplicateQueryStringsDetectorImpl.java]: i dont think we need to make this change at all.
at first i was going to suggest returning early if _resultRankingsConfiguration.allowDuplicateQueryStrings() is true (see above comment on line 55), but then i checked the usages of detect in EditRankingMVCActionCommand, and it looks like we already return early https://github.com/BryanEngler/liferay-portal/blob/29b4293a37b899069241d36a53adc09ccd3c57d5/modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/main/java/com/liferay/portal/search/tuning/rankings/web/internal/portlet/action/EditRankingMVCActionCommand.java#L287
https://github.com/BryanEngler/liferay-portal/blob/29b4293a37b899069241d36a53adc09ccd3c57d5/modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/main/java/com/liferay/portal/search/tuning/rankings/web/internal/portlet/action/EditRankingMVCActionCommand.java#L300
so the logic here would never even be hit.

Comment (thektan) [reply modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/main/java/com/liferay/portal/search/tuning/rankings/web/internal/index/DuplicateQueryStringsDetectorImpl.java]: Ah I see...so if we want inactive rankings to be checked for duplicates too we'd just need to remove `queries.term(RankingFields.INACTIVE, true)?

Comment (thektan) [reply modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/main/java/com/liferay/portal/search/tuning/rankings/web/internal/index/DuplicateQueryStringsDetectorImpl.java]: I'll take a look at this again, since while testing it out I was able to create duplicate rankings as long as they were inactive, but then I was able to activate duplicate rankings afterwards.

So maybe duplicate inactive rankings are fine as long as you can only activate one at a time.

Comment (thektan) [reply modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/main/java/com/liferay/portal/search/tuning/rankings/web/internal/index/DuplicateQueryStringsDetectorImpl.java]: Got it. I'll drop these changes and add `guardDuplicateQueryString(actionRequest, editRankingMVCActionRequest);` to `com.liferay.portal.search.tuning.rankings.web.internal.portlet.action.EditRankingMVCActionCommand#doUpdate`.

**Comment by thektan** [comment]
@BryanEngler Updated! Dropped the changes in `DuplicateQueryStringsDetectorImpl.java` and added https://github.com/BryanEngler/liferay-portal/pull/435/commits/d0d13c2dc7b38a0da6a05e8a9d268cec8d2bbc71

**Comment by BryanEngler** [comment]
ya looks good to me, just curious, do you know what happens if duplicates are allowed and there are a bunch of active duplicates, and then you go to the config and turn off duplicates? from what ive seen, it seems like that would only prevent future adds/updates from allowing duplicates, but the existing duplicates would still be active/allowed

**Comment by BryanEngler** [comment]
ci:test:search:87ddb7a080cf78839171279da79914beca511050

**Comment by thektan** [comment]
Ya, the existing duplicates would stay active. When you edit one of those duplicate active rankings, you probably won't be able to save though...

**Comment by thektan** [comment]
I realized the `inactive` url param value also needs to be passed from the action menu. 😕 I'll wait for the ci tests to finish before pushing up the change.

**Comment by thektan** [comment]
Sorry for this not being as straightforward as I thought. 😓 

Turns out guarding query strings prevents updating any ranking since it doesn't exclude itself while looking for dupes. I pushed up the change 
7bd7e8b and we'll have to run the tests again.

**Comment by thektan** [comment]
ci:test:search:87ddb7a080cf78839171279da79914beca511050

**Comment by thektan** [comment]
Added 634a406 to always allow a ranking to be set to inactive for that edge case where the allow duplicate config is set to true and then false later.

**Comment by thektan** [comment]
ci:test:search:87ddb7a080cf78839171279da79914beca511050

**Comment by BryanEngler** [comment]
i added a commit to remove duplicate logic, testing here https://github.com/BryanEngler/liferay-portal/pull/438
